core/domain/user_services.py

```diff
@@ -713,8 +716,27 @@ def get_usernames(user_ids):
         If a user_id does not exist, the corresponding entry in the
         returned list is None.
     """
-    users_settings = get_users_settings(user_ids)
-    return [us.username if us else None for us in users_settings]
+    usernames = [''] * len(user_ids)
```

Comment:
Might as well set to None now, and then you don't need the else clause below.

Reply:
Done.